manual_code/lab2/TM/src/kernel/k_task.c

```diff
@@ -389,9 +389,14 @@ int k_tsk_create(task_t *task, void (*task_entry)(void), U8 prio, U16 stack_size
 
 void k_tsk_exit(void) 
 {
-#ifdef DEBUG_0
-    printf("k_tsk_exit: entering...\n\r");
-#endif /* DEBUG_0 */
+// #ifdef DEBUG_0
+//     printf("k_tsk_exit: entering...\n\r");
+// #endif /* DEBUG_0 */
+    gp_current_task->state = DORMANT;
+    
```

Comment:
Don't forget to decrement g_num_active_tasks

Reply:
updated